src/odemis/acq/stitching/_weaver.py

```diff
@@ -57,6 +57,10 @@ def addTile(self, tile):
         """
         # Merge the correction metadata inside each image (to keep the rest of the
         # code simple)
+        if isinstance(tile, tuple):
+            tile = tile[0]
+        if isinstance(tile, model.DataArrayShadow):
```

Comment:
Moved the data load to the caller, added a type check explaining how to convert from DataArrayShadow -> DataArray. I think it would be better if it just loaded it automatically, but agree should be consistent with other functionality